.eslintrc.js

```diff
@@ -49,6 +51,76 @@ const config = {
+          ' > JSXFragment > .children:not(JSXElement, JSXText[value=/^\\s+$/])',
+        message:
+          'Conditional plain text nodes could break React if used with Google Translate. Wrap text into an element.',
+      },
+      // Ban text nodes before or after a condition `text {condition && <span/>} text`
+      {
+        selector:
+          'JSXElement:has(JSXExpressionContainer.children > LogicalExpression[operator="&&"])' +
+          ' > JSXText[value!=/^\\s+$/]',
+        message:
+          'Plain text nodes before or after a condition could break React if used with Google Translate. Wrap text into an element.',
+      },
+      // Ban variables before or after `{var} {condition && <span/>} {var}` (just in case they return a string)
+      {
+        selector:
+          'JSXElement:has(JSXExpressionContainer.children > LogicalExpression[operator="&&"])' +
+          ' > JSXExpressionContainer:matches([expression.type="Identifier"], [expression.type="CallExpression"])',
+        message:
+          'Plain text nodes before or after a condition could break React if used with Google Translate. Identifier could possibly return a string, so wrap it into an element.',
+      },
+
+      // Ban `condition ? text node : <span/>`
+      {
+        selector:
+          'JSXElement > JSXExpressionContainer > ConditionalExpression' +
+          ' > :matches(.consequent, .alternate):not(JSXElement, JSXFragment)',
+        message:
+          'Conditional plain text nodes could break React if used with Google Translate. Wrap text into an element.',
+      },
+      // Ban `condition ? <>text node</> : <span/>`
+      {
+        selector:
+          'JSXElement > JSXExpressionContainer > ConditionalExpression' +
+          ' > JSXFragment > .children:not(JSXElement, JSXText[value=/^\\s+$/])',
+        message:
+          'Conditional plain text nodes could break React if used with Google Translate. Wrap text into an element.',
+      },
+      // Ban text nodes before or after a condition `text {condition ? <span/> : <span/>} text`
+      {
+        selector:
+          'JSXElement:has(JSXExpressionContainer.children > ConditionalExpression)' +
+          ' > JSXText[value!=/^\\s+$/]',
+        message:
+          'Plain text nodes before or after a condition could break React if used with Google Translate. Wrap text into an element.',
+      },
+      // Ban variables before or after `{var} {condition ? <span/> : <span/>} {var}` (just in case they return a string)
+      {
+        selector:
+          'JSXElement:has(JSXExpressionContainer.children > ConditionalExpression)' +
+          ' > JSXExpressionContainer:matches([expression.type="Identifier"], [expression.type="CallExpression"])',
+        message:
+          'Plain text nodes before or after a condition could break React if used with Google Translate. Identifier could possibly return a string, so wrap it into an element.',
+      },
```

Comment:
This probably covers most potential issues, but there is a possibility of undetectable:
```tsx
const a = condition ? 'aa' : <span>bb</span>
return <Box>{a}</Box>
```

Reply:
We should later make a package out of it, and reuse it on explorer